### PR TITLE
Add jib

### LINKED ISSUE #424: Build Docker Image
I don't know what is the best solution :
- use JIB
- add a Dockerfile at root folder (which will contain the build part too)

Maybe we can provide 2 options

### PR DESCRIPTION
**Pull request by swarajsaaj**
Fix #424
Add jib through a common docker context

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Thanks @swarajsaaj !
I'll review it soon ;-)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/docker/infrastructure/primary/rest/SpringBootDockerResource.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import tech.jhipster.lite.generator.project.domain.Project;
+import tech.jhipster.lite.generator.project.infrastructure.primary.dto.ProjectDTO;
+import tech.jhipster.lite.generator.server.springboot.docker.application.SpringBootDockerApplicationService;
+
+@RestController
+@RequestMapping("/api/servers/spring-boot/docker")
+@Tag(name = "Spring Boot")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/docker/infrastructure/primary/rest/SpringBootDockerResource.java:16]: I suggest to move to tag: Spring Boot - Tools

Then, I'll move async, devtools under this tag too

File: src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+              <image>%s:latest</image>
+            </to>
+            <container>
+              <entrypoint>
+                <shell>bash</shell>
+                <option>-c</option>
+                <arg>/entrypoint.sh</arg>
+              </entrypoint>
+              <ports>
+                <port>8081</port>
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java:67]: the port should not be 8081, but the same one in application.properties, so it's the property `serverPort`

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
Few comments. Otherwise, I tested it and it works perfectly :)
good job

File: src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java (hunk 2 above)
Comment (swarajsaaj) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java:67]: @pascalgrimaud Shall I move `getServerPort` https://github.com/jhipster/jhipster-lite/blob/6243a3253b136b84a5c36d9068997fca3a90cd00/src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/web/domain/SpringBootMvcDomainService.java#L125 to `Project` class for common access?

Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java:67]: Yes I think it's a good idea. Possible to do it, in a separate PR first?

Comment (swarajsaaj) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java]: Done prposed change  in [c7ad56](https://github.com/jhipster/jhipster-lite/pull/554/commits/c7ad56095c43f3435790095ea19bf53042428430)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/docker/infrastructure/primary/rest/SpringBootDockerResource.java (hunk 1 above)
Comment (swarajsaaj) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/infrastructure/primary/rest/SpringBootDockerResource.java]: Sure. done

File: src/test/java/tech/jhipster/lite/generator/project/domain/ProjectTest.java [hunk 3]
```diff
@@ -344,6 +344,34 @@ void shouldNotGetIntegerConfig() {
...
+      assertThat(project.getServerPort()).isEqualTo(1337);
+    }
+
+    @Test
+    void shouldNotGetServerPort() {
+      Project project = tmpProject();
+
+      project.addConfig("serverPort", List.of(1337));
+
+      assertThatThrownBy(() -> project.getIntegerConfig("serverPort"))
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/project/domain/ProjectTest.java:369]: getServerPort() instead

Comment (swarajsaaj) [reply src/test/java/tech/jhipster/lite/generator/project/domain/ProjectTest.java]: My bad, infact exception is never thrown, instead default 8080 is returned. Thanks for noticing (https://github.com/jhipster/jhipster-lite/pull/554/commits/83f5d228799d2a51a9ec89258fb5052e5977502e)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java (hunk 2 above)
Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java]: as you moved the code, can you clean these import too plz ?

image

After that, it should be OK to be merged :)

Comment (swarajsaaj) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java]: Removed, should we add the sonar quality gate in actions too as breaking step, is it possible for PRs in sonarcloud?

Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/docker/domain/SpringBootDockerDomainService.java]: See https://github.com/jhipster/jhipster-lite/pull/374/files

I tried to introduce code smell, but Sonarcloud didn't detect it...

**Pull request merged**